cluster/mesos/docker/util.sh

```diff
@@ -0,0 +1,312 @@
+  "${kubectl}" config use-context "${CONTEXT}" --cluster="${CONTEXT}"
+
+   echo "Wrote config for ${CONTEXT} to ${KUBECONFIG}" 1>&2
+}
+
+# Perform preparations required to run e2e tests
+function prepare-e2e {
+  echo "TODO: prepare-e2e" 1>&2
+}
+
+# Execute prior to running tests to build a release if required for env
+function test-build-release {
+  # Make a release
+  export KUBERNETES_CONTRIB=mesos
+  export KUBE_RELEASE_RUN_TESTS=N
+  "${KUBE_ROOT}/build/release.sh"
+}
+
+# Must ensure that the following ENV vars are set
+function detect-master {
+  #  echo "KUBE_MASTER: $KUBE_MASTER" 1>&2
+
+  docker_id=$(docker ps --filter="name=docker_apiserver" --quiet)
+  if [[ "${docker_id}" == *'\n'* ]]; then
+    echo "ERROR: Multiple API Servers running in docker" 1>&2
+    return 1
+  fi
+
+  details_json=$(docker inspect ${docker_id})
+  master_ip=$(jq '.[0].NetworkSettings.IPAddress' --raw-output <<< "$details_json")
+  master_port=$(jq '.[0].NetworkSettings.Ports[][].HostPort' --raw-output <<< "$details_json")
+
+  KUBE_MASTER_IP="${master_ip}:${master_port}"
+  KUBE_SERVER="http://${KUBE_MASTER_IP}"
+
+  echo "KUBE_MASTER_IP: $KUBE_MASTER_IP" 1>&2
+}
+
+# Get minion IP addresses and store in KUBE_MINION_IP_ADDRESSES[]
+# These Mesos slaves MAY host Kublets,
+# but might not have a Kublet running unless a kubernetes task has been scheduled on them.
+function detect-minions {
+  docker_ids=$(docker ps --filter="name=docker_mesosslave" --quiet)
+  while read -r docker_id; do
+
+    details_json=$(docker inspect ${docker_id})
+    minion_ip=$(jq '.[0].NetworkSettings.IPAddress' --raw-output <<< "$details_json")
+    minion_port=$(jq '.[0].NetworkSettings.Ports[][].HostPort' --raw-output <<< "$details_json")
+    #TODO: filter for 505* port
+
+    KUBE_MINION_IP_ADDRESSES+=("${minion_ip}")
+#    KUBE_MINION_IP_ADDRESSES+=("${minion_ip}:${minion_port}")
+
+  done <<< "$docker_ids"
+  echo "KUBE_MINION_IP_ADDRESSES: [${KUBE_MINION_IP_ADDRESSES[*]}]" 1>&2
+}
+
+# Verify prereqs on host machine
+function verify-prereqs {
+  echo "TODO: verify-prereqs" 1>&2
+  # Verify that docker, docker-compose, and jq exist
+  # Verify that all the right docker images exist:
+  # mesosphere/kubernetes-mesos-test, etc.
+}
+
+# create the apiserver key and certificate with the now known
+# ip address, 10.10.10.1 and some kubernetes.* domain names in
+# the subjectAltNames field
+function cluster::mesos::docker::create_apiserver_cert {
+  local APISERVER_IP="$1"
+  local SERVICE_IP="$2"
+
+  # find openssl.cnf
+  local OPENSSL_CNF=$(for CANDIDATE in "/etc/ssl/openssl.cnf" "/System/Library/OpenSSL/openssl.cnf"; do
+      if [ -f "$CANDIDATE" ]; then
+        echo $CANDIDATE
+        break
+      fi
+  done)
+  if [ -z "$OPENSSL_CNF" ]; then
+    echo "ERROR: cannot find openssl.cnf"
+    exit 2
+  fi
+
+  # create apiserver key and certificate sign request
+  workspace=$(env TMPDIR=$PWD mktemp -d -t "k8sm-workspace-XXXXXX")
```

Comment:
perhaps use "k8sm-cert-workspace-XXXXXX" as the prefix to avoid overlapping. I also made a nifty run_in_temp_dir function you might want to use instead.